dashboard/assets/components/Dashboard.jsx

```diff
@@ -0,0 +1,146 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {withStyles} from 'material-ui/styles';
+
+import SideBar from './SideBar.jsx';
+import Header from './Header.jsx';
+import Main from "./Main.jsx";
+import {isNullOrUndefined, defaultZero, MEMORY_SAMPLE_LIMIT, TAGS} from "./Common.jsx";
+
+// Styles for the Dashboard component.
+const styles = theme => ({
+    appFrame: {
+        position: 'relative',
+        display: 'flex',
+        width: '100%',
+        height: '100%',
+        background: '#303030',
+    },
+});
+
+// Dashboard is the main component, which renders the whole page,
+// makes connection with the server and listens for messages.
+// When there is an incoming message, updates the page's content correspondingly.
+class Dashboard extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            active: TAGS.home.id, // active menu
+            sideBar: true, // true if the sidebar is opened
+            memory: [],
+            traffic: [],
+            logs: [],
+        };
+    }
+
+    // componentDidMount initiates the establishment of the first websocket connection after the component is rendered.
+    componentDidMount() {
+        this.reconnect();
+    }
+
+    // reconnect establishes a websocket connection with the server, listens for incoming messages
+    // and tries to reconnect on connection loss.
+    reconnect = () => {
+        const server = new WebSocket("ws://" + location.host + "/api");
+
+        server.onmessage = event => {
+            const msg = JSON.parse(event.data);
+            if (isNullOrUndefined(msg)) {
+                return;
+            }
+            this.update(msg);
+        };
+
+        server.onclose = () => {
+            setTimeout(this.reconnect, 3000);
+        };
+    };
+
+    // update analyzes the incoming message, and updates the charts' content correspondingly.
+    update = msg => {
+        // (Re)initialize the state with the past data. metrics is set only in the first msg,
+        // after the connection is established.
+        if (!isNullOrUndefined(msg.metrics)) {
+            let memory = [];
+            let traffic = [];
+            if (!isNullOrUndefined(msg.metrics.memory)) {
+                memory = msg.metrics.memory.map(elem => ({memory: elem.value}));
+                traffic = msg.metrics.processor.map(elem => ({traffic: defaultZero(elem.value)})) // TODO (kurkomisi): traffic != processor!!!
+            }
```

Comment:
Please retain data arrays here and only convert later when actually rendering. Also I don't think there's a need to have specialized `memory` and `traffic` datakeys, we can just use `value` as the key (my 2c, open for debate).